File: fibonacci/clojure/fibonacci.clj
(ns fibonacci)

(set! *unchecked-math* :warn-on-boxed)

(definterface IFib
  (^long fib [^long n]))

(deftype Fibonacci []
  IFib
  (fib [_  n]
    (if (< n 2)
      (long n)
      (long (+ (.fib _ (- n 1))
               (.fib _ (- n 2)))))))

(def ^Fibonacci fib (Fibonacci.))

File: fibonacci/zig/fibonacci.zig
pub fn fibonacci(comptime T: type) fn (T) T {
    return struct {
        pub fn fib(n: T) T {
            return switch (n) {
                0...1 => n,
                else => fib(n - 1) + fib(n - 2),
            };
        }
    }.fib;
}
